zebra/zebra_rib.c

```diff
@@ -1334,19 +1334,15 @@ static bool rib_update_re_from_ctx(struct route_entry *re,
 	 * we use the fib-specific nexthop-group to record the actual FIB
 	 * status.
 	 */
+	matched = false;
+	ctxnhg = dplane_ctx_get_ng(ctx);
 
 	/* Check both fib group and notif group for equivalence.
 	 *
 	 * Let's assume the nexthops are ordered here to save time.
 	 */
-	if (nexthop_group_equal(&re->fib_ng, dplane_ctx_get_ng(ctx)) == false) {
-		if (IS_ZEBRA_DEBUG_RIB_DETAILED)
-			zlog_debug(
-				"%s(%u):%s update_from_ctx: notif nh and fib nh mismatch",
-				VRF_LOGNAME(vrf), re->vrf_id, dest_str);
-
-		matched = false;
-	} else
+	if (re->fib_ng.nexthop && ctxnhg->nexthop &&
```

Comment:
How is this fixing it exactly? The logic looks the same as it was before minus the debug. Maybe I am missing something?

Reply:
the issue we found was that "empty" was equal to "empty" - so the code didn't act when that particular condition happened. Now, the code is confirming that there's something to compare, and _then_ comparing.